src/indexes/text/term.h

```diff
@@ -54,6 +54,12 @@ class TermIterator : public TextIterator {
   const PositionRange& CurrentPosition() const override;
   bool NextPosition() override;
   FieldMaskPredicate CurrentFieldMask() const override;
+  // Returns true if iterator is at a valid state with current key, position,
+  // and field.
+  bool IsIteratorValid() const override {
+    return !DoneKeys() && current_position_.has_value() &&
```

Comment:
Why do we call DoneKeys? That is not a performant API. We can simply check if the current key is set, right?